README.md

```diff
@@ -297,6 +302,76 @@ If you are using IAM role configuration (`iam-role`) for EC2 discovery, you need
+
+### AWS Autoscaling using Lifecycle Hooks
+
+This solution is recommended and safe, no matter what your cluster and data sizes are. Nevertheless, if you find it too complex, you may want to try alternative solutions.
+
+#### AWS Autoscaling Architecture
+
+The necessary autoscaling architecture is presented on the diagram.
+
+![architecture](markdown/images/aws-autoscaling-architecture.png)
+
+The following activities must be performed to set up the Hazelcast AWS Autoscaling process:
+
+* Create **AWS SQS** queue
+* Add **`lifecycle_hook_listener.sh`** to each instance
+  * The `lifecycle_hook_listener.sh` script can be started as the User Data script
+  * Alternatively, it's possible to create a separate dedicated EC2 Instance running only `lifecycle_hook_listener.sh` in the same network and leave the autoscaled EC2 Instances only for Hazelcast members
+* Set `Scaling Policy` to `Step scaling` and increase/decrease always by adding/removing `1 instance`
+* Create `Lifecycle Hooks`
+  * Instance Launch Hook
+  * Instance Terminate Hook
+
+#### Lifecycle Hook Listener Script
+
+The `lifecycle_hook_listener.sh` script takes one argument as a parameter: `queue_name` (AWS SQS name). It performs operations that can be expressed in the following pseudocode.
+
+```
+while true:
+    message = receive_message_from(queue_name)
+    instance_ip = extract_instance_ip_from(message)
+    while not is_cluster_safe(instance_ip):
+        sleep 5
+    send_continue_message
+```
+
+### AWS Autoscaling Alternative Solutions
+
+The solutions below are not recommended, since they may not operate well under certain conditions. Therefore, please use them with caution.
+
+#### Cooldown Period
+
+[Cooldown Period](https://docs.aws.amazon.com/autoscaling/ec2/userguide/Cooldown.html) is a statically defined time interval that AWS Autoscaling Group waits before the next autoscaling operation takes place. If your cluster is small and predictable, then you can use it instead of Lifecycle Hooks. 
+
+* Set `Scaling Policy` to `Step scaling` and increase/decrease always by adding/removing `1 instance`
+* Set `Cooldown Period` to a reasonable value (which depends on the cluster and data size)
+
+Note the drawbacks:
+
+ * If the cluster contains a significant amount of data, it may be impossible to define one static cooldown period
+ * Even if the cluster comes back to the safe state quicker, the next operation needs to wait the defined cooldown period
+
+#### Graceful Shutdown
+
+A solution that may sound simple and good (but is actually not recommended) is to use **[Hazelcast Graceful Shutdown](http://docs.hazelcast.org/docs/latest-development/manual/html/FAQ.html#page_How+can+I+shutdown+a+Hazelcast+member)** as a hook on the EC2 Instance Termination. In other words, without any Autoscaling-specific features (like Lifecycle Hooks), you could adapt the EC2 Instance to wait for the Hazelcast member to shutdown before terminating the instance.
```

Comment:
I think it is better provide a fixed version instead of latest develoment. 
http://docs.hazelcast.org/docs/3.9.4/manual/html-single/index.html#how-can-i-shutdown-a-hazelcast-member

Reply:
Totally agree. Changed. Thanks!